components/builder-api/src/depot/server.rs

```diff
@@ -1426,6 +1444,16 @@ fn package_privacy_toggle(req: &mut Request) -> IronResult<Response> {
         None => return Ok(Response::with(status::BadRequest)),
     };
 
+    // This stinks but is necessary since memcached Client returns mutable reference
+    // and iron chain.link returns everything wrapped in Arc
+    // The tradeoff here is that we have to initialize a new client on every request until
+    // we can switch to a framework that's not Iron
+    let depot = req.get::<persistent::Read<Config>>().unwrap();
+    let memcache_host_strings: Vec<String> =
+        depot.memcached.iter().map(|h| format!("{}", h)).collect();
+    let memcache_hosts: Vec<&str> = memcache_host_strings.iter().map(AsRef::as_ref).collect();
+    let mut client = memcache::Client::new(memcache_hosts).unwrap();
```

Comment:
This ugliness is repeated a few times.  You could make a helper function that takes an iron `req` and returns a `memcache::Client`, if you want.

Reply:
Alternatively, just have the _memcache_package functions create the client internally so the callers don't have to do it.

Reply:
That's actually a better idea 👍